openassessment/xblock/ui_mixins/mfe/page_context_serializer.py

```diff
@@ -237,58 +299,46 @@ class PageDataSerializer(Serializer):
     Requires context to differentiate between Assessment and Submission views
     """
 
-    require_context = True
+    requires_context = True
 
     progress = ProgressSerializer(source="*")
-    submission = SerializerMethodField()
-    rubric = RubricConfigSerializer(source="*")
+    response = SerializerMethodField()
     assessment = SerializerMethodField()
 
     def to_representation(self, instance):
-        # Loading workflow status causes a workflow refresh
-        # ... limit this to one refresh per page call
-        if not self.context.get("step"):
-            active_step = instance.workflow_data.status or "submission"
-            self.context.update({"step": active_step})
+        if "step" not in self.context:
+            raise ValidationError("Missing required context: step")
+        if "view" not in self.context:
+            raise ValidationError("Missing required context: view")
 
         return super().to_representation(instance)
 
-    def _can_jump_to_step(self, workflow_step, workflow_data, step_name):
-        """
-        Helper to determine if a student can jump to a specific step:
-        1) Student is on that step.
-        2) Student has completed that step.
-
-        NOTE that this should probably happen at the handler level, but for
-        added safety, check here as well.
-        """
-        if step_name == workflow_step:
-            return True
-        step_status = workflow_data.status_details.get(step_name, {})
-        return step_status.get("complete", False)
-
-    def get_submission(self, instance):
+    def get_response(self, instance):
         """
         we get the user's draft / complete submission.
         """
         # pylint: disable=broad-exception-raised
+
         # Submission Views
         if self.context.get("view") == "submission":
             learner_page_data_submission_data = instance.get_learner_submission_data()
```

Comment:
since we've changed the name of the serializer, we can also change the name of this variable